ufl/utils/sorting.py

```diff
@@ -84,6 +86,8 @@ def canonicalize_metadata(metadata):
             value = str(value)
         elif hasattr(value, "ufl_signature"):
             value = value.ufl_signature
+        elif isinstance(value, np.ndarray):
+            value = np.array2string(value, separator=" ")
```

Comment:
This is what happens for a numpy array as input to metadata atm irregardless (just throws a user warning).

Reply:
Sure, I'm just nervous about silencing a warning that might have an actual point.

Reply:
If we want to avoid people sending in large arrays of custom quadrature points/weights we should find a better way to let people patch in custom quadrature rules.

Currently they have to be implemented in FINAT/FIAT/basix?

Reply:
Let's discuss this in person tomorrow. I have ideas.